arccnet/data_generation/timeseries/sdo_processing.py

```diff
@@ -269,30 +333,64 @@ def hmi_query_export(time_1, time_2, keys: list, sample: int):
 
     Returns
     -------
-        hmi_query_full, hmi_result : `tuple`
-            A tuple containing the query result (pandas df) and the export data response (drms export object).
+        hmi_query_full, hmi_result, ic_query_full, ic_export : `tuple`
+            A tuple containing the query results of the hmi mag and ic_no_limbdark (pandas df) and the export data response (drms export object).
     """
     client = drms.Client()
     duration = round((time_2 - time_1).to_value(u.hour))
-    qstr_hmi = f"hmi.M_720s[{time_1.value}/{duration}h@{sample}m]" + "{magnetogram}"
-    hmi_query = client.query(ds=qstr_hmi, key=keys)
+    qstr_m_hmi = f"hmi.M_720s[{time_1.value}/{duration}h@{sample}m]" + "{magnetogram}"
+    hmi_query = client.query(ds=qstr_m_hmi, key=keys)
 
     good_result = hmi_query[hmi_query.QUALITY == 0]
     good_num = good_result["*recnum*"].values
     bad_result = hmi_query[hmi_query.QUALITY != 0]
 
-    qstrs_hmi = [f"hmi.M_720s[{time}]" + "{magnetogram}" for time in bad_result["T_REC"]]
-    hmi_values = [hmi_rec_find(qstr, keys) for qstr in qstrs_hmi]
+    qstrs_m_hmi = [f"hmi.M_720s[{time}]" + "{magnetogram}" for time in bad_result["T_REC"]]
+    hmi_values = [hmi_rec_find(qstr, keys, 3, 720) for qstr in qstrs_m_hmi]
     patched_num = [*hmi_values]
 
     joined_num = [*good_num, *patched_num]
     joined_num = [str(num) for num in joined_num]
     hmi_num_str = str(joined_num).strip("[]")
+
     hmi_qstr = f"hmi.M_720s[! recnum in ({hmi_num_str}) !]" + "{magnetogram}"
     hmi_query_full = client.query(ds=hmi_qstr, key=keys)
     hmi_result = client.export(hmi_qstr, method="url", protocol="fits", email=os.environ["JSOC_EMAIL"])
     hmi_result.wait()
-    return hmi_query_full, hmi_result
+    ic_query_full, ic_result = hmi_continuum_export(hmi_query_full, keys)
+    return hmi_query_full, hmi_result, ic_query_full, ic_result
+
+
+def hmi_continuum_export(hmi_query, keys):
+    r"""
+    Query and export HMI continuum data from the JSOC database.
+
+    Parameters
+    ----------
+        hmi_query : `drms query`
+            HMI query result containing target times.
+        keys : `list`
+            A list of keys to query.
+
+    Returns
+    -------
+        hmi_query_full, hmi_result, ic_query_full, ic_export : `tuple`
+            A tuple containing the query results of the hmi mag and ic_no_limbdark (pandas df) and the export data response (drms export object).
+    """
+    client = drms.Client()
+    qstrs_ic = [f"hmi.Ic_noLimbDark_720s[{time}]" + "{image}" for time in hmi_query["T_REC"]]
```

Comment:
Why is the segment image here but continuum in the query just curious?

You can escape curly bracket by doubling them in f-strings `f"hmi.Ic_noLimbDark_720s[{time}]{{image}}"`